### ISSUE TITLE
Enhancement: Quiet Output

### ISSUE THREAD

**Issue by TheDrawingCoder-Gamer** [HaxeFoundation/haxe#10408]
I mostly want this for haxelib, but make it so we can have a quiet flag when compiling.
Best use for this imo is haxelib adding the flag so it doesn't spit out garbage when running an app that has warnings.

**Comment by kLabz** [comment]
Check [`haxe.macro.Context.filterMessages()`](https://github.com/HaxeFoundation/haxe/blob/development/std/haxe/macro/Context.hx#L95) (a little more context on #8471)

**Comment by TheDrawingCoder-Gamer** [comment]
this implies you can disable it with macros, right?

**Comment by kLabz** [comment]
Yep

**Comment by TheDrawingCoder-Gamer** [comment]
I'd prefer a direct edit to the compiler as it would be more helpful there instead of using macros. Unsure if I should close this because it's technically been solved but not in a very nice way.

**Comment by kLabz** [comment]
Yeah, could be useful to hook some basic behavior as compiler flags.

Meanwhile, a quick alternative:

```
haxelib install no-warnings
haxe [... your args ...] -lib no-warnings
```

**Comment by TheDrawingCoder-Gamer** [comment]
semicursed but still good. 

leaving this issue open as an "Add compiler flags please" issue

**Comment by RealyUniqueName** [comment]
I'm all for reducing noise from warnings, but as I said in #6725 I don't like hiding problems completely and I would not use `--no-warnings` option. 
However I'm leaving this issue open since it's kind of "industry standard" to have such flag.

**Comment by kLabz** [comment]
Did you implement your warnings summary somewhere btw?

**Comment by RealyUniqueName** [comment]
Only in #6725. 
By the time you submitted your #8471 I already had no projects with warnings issues.
